### PR TITLE
Add login page

### PR DESCRIPTION
**Pull request by Serneum**
Only the last commit needs to be reviewed

Helps complete #5 

Log In page:
image

Registration page (note Log In link at the bottom):
image

When logged in:
image

### CODE REVIEW

File: lib/agoneum/account/session.ex [hunk 1]
```diff
@@ -14,6 +14,12 @@ defmodule Agoneum.Account.Session do
   end
   def authenticate(_), do: {:error, :invalid_credentials}
 
+  def current_user(conn) do
+    Guardian.Plug.current_resource(conn)
+  end
+
+  def logged_in?(conn), do: !!current_user(conn)
```
Comment (mbramson) [inline lib/agoneum/account/session.ex:21]: It's a matter of choice but do you really need this? Guardian.Plug.current_resource either returns nil or a User, so you can condition on it just the same.

File: lib/agoneum/web/views/error_helpers.ex [hunk 2]
```diff
@@ -9,9 +9,9 @@ defmodule Agoneum.Web.ErrorHelpers do
   Generates tag for inlined form input errors.
   """
   def error_tag(form, field) do
-    if error = form.errors[field] do
+    Enum.map(Keyword.get_values(form.errors, field), fn (error) ->
```
Comment (Serneum) [inline lib/agoneum/web/views/error_helpers.ex]: This came from [Phoenix](https://github.com/phoenixframework/phoenix/blob/v1.3.0-rc.2/installer/templates/new/web/views/error_helpers.ex#L11-L15)

File: lib/agoneum/web/views/layout_view.ex [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+  end
+  defp user_display_name(user) when not is_nil(user) do
+    user_display_name(user.email, user.name)
+  end
+  defp user_display_name(_) do
+    "Guest"
+  end
+  defp user_display_name(_, name) when not is_nil(name) do
+    name
+  end
```
Comment (mbramson) [inline lib/agoneum/web/views/layout_view.ex]: I think you could avoid the guard clause by having `user_display_name(email, nil)` above `user_display_name(_, name)`, but I think that's honestly a matter of preference.

**Review by mbramson — APPROVED** [review]
A few random thoughts but overall this LGTM

File: lib/agoneum/web/views/layout_view.ex (hunk 3 above)
Comment (Serneum) [reply lib/agoneum/web/views/layout_view.ex]: Good call

File: lib/agoneum/account/session.ex (hunk 1 above)
Comment (Serneum) [reply lib/agoneum/account/session.ex:21]: I'm gonna leave this because it's more readable/obvious

**Pull request merged**